src/plugins/discover/public/application/main/components/layout/discover_documents.tsx

```diff
@@ -261,6 +262,10 @@ function DiscoverDocumentsComponent({
   const { customCellRenderer, customGridColumnsConfiguration } =
     useContextualGridCustomisations() || {};
 
+  const baseGetCellRenderers = useCallback(() => customCellRenderer ?? {}, [customCellRenderer]);
+  const getCellRenderers = useProfileAccessor('getCellRenderers', baseGetCellRenderers);
+  const cellRenderers = useMemo(() => getCellRenderers(), [getCellRenderers]);
```

Comment:
I attempted this for a while, but I couldn't figure out how to implement a hook similar to the suggestion without breaking memoization, and because the accessed method may accept parameters. Definitely open to thoughts if there's something I'm missing though.

I did manage to simplify the hook a bit so only two declarations are needed for each use though: https://github.com/elastic/kibana/pull/183797/commits/f4cb0f3826933f1b2a8acb1c7837b530ae54bc01.